### PR TITLE
2020 09 07 small fixes

### PR DESCRIPTION
**Pull request by tomrittervg**
Okay, this is tested and ready. (It should pass at least....)

A bunch of assorted work that handles some MVP issues and some v1/vFuture things that were needed or related.

### CODE REVIEW

File: localconfig.py.example [hunk 1]
```diff
@@ -15,8 +15,8 @@ localconfig = {
     },
     'Database': {
         'host': 'localhost',
-        'user': 'username',
-        'password': 'p4ssw0rd',
+        'user': 'root',
```
Comment (metalcanine) [inline localconfig.py.example:18]: 😎

File: tests/functionality.py [hunk 2]
```diff
@@ -193,6 +193,10 @@ def _setup(try_revision, library_filter):
     def _cleanup(u, expected_values):
         for l in u.dbProvider.get_libraries():
             u.dbProvider.delete_job(l, expected_values.library_version_id)
+        # Needed so we do a database cleanup and prevent locking in MySQL.
```
Comment (metalcanine) [inline tests/functionality.py:196]: Should we add a TODO: comment to come back to this?

File: apis/taskcluster.py [hunk 3]
```diff
@@ -7,17 +7,17 @@
...
 import jsone
 import requests
+from urllib.parse import quote_plus
 
 from components.utilities import Struct
 from components.logging import logEntryExit, logEntryExitNoArgs, LogLevel
 from components.providerbase import BaseProvider, INeedsCommandProvider, INeedsLoggingProvider
 
-RETRIGGER_NUMBER = 3
+RETRIGGER_TOTAL = 4
```
Comment (metalcanine) [inline apis/taskcluster.py:16]: Can we add a comment here about why we picked 4 so that we don't unintentionally bump it higher in the future?

**Review by metalcanine — APPROVED** [review]
Just a few small nits, but looks great to me!

File: .circleci/config.yml [hunk 4]
```diff
@@ -7,9 +7,8 @@ jobs:
       - image: circleci/mysql:8.0.19
         command: [--default-authentication-plugin=mysql_native_password]
         environment:
+          MYSQL_ROOT_PASSWORD: rootpw
           MYSQL_DATABASE: updatebot
```
[the remark this answers is not in the document]
Comment (tomrittervg) [reply .circleci/config.yml:11]: I don't think that will work, if we don't have a password set, the db will be uninitialized and updatebot won't be able to connect:
https://github.com/docker-library/mysql/blob/285fd39122e4b04fbe18e464deb72977df9c6a45/8.0/docker-entrypoint.sh#L137

File: tests/functionality.py (hunk 2 above)
Comment (tomrittervg) [reply tests/functionality.py:196]: I think I can remove this now...

**Comment by tomrittervg** [comment]
@metalcanine Okay, assuming tests pass, I think I addressed comments.

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (metalcanine) [reply .circleci/config.yml:11]: Sorry, I should have been less vague and github's UI isn't particularly helpful here :/ I meant to say we could try removing line 11 `MYSQL_DATABASE: updatebot`

**Pull request merged**